src/App.jsx

```diff
@@ -14,57 +15,62 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [goodText, setGoodText] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {goodText.length ? null : `No goods selected`}
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+      <h1 className="title is-flex is-align-items-center">
+        {goodText.length ? `${goodText} is selected` : null}
+        {goodText.length ? (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              setGoodText('');
+            }}
+          />
+        ) : null}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              className={
+                goodText === good ? 'has-background-success-light' : null
+              }
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={goodText === good ? `RemoveButton` : `AddButton`}
+                  type="button"
+                  className={`button ${goodText === good ? `is-info` : null}`}
+                  onClick={() => {
+                    if (goodText === good) {
+                      setGoodText('');
+                    } else {
+                      setGoodText(good);
+                    }
+                  }}
```

Comment:
Move this logic to handler function. Also you can simplify it a bit
```suggestion
                    if (goodText === good) {
                      setGoodText('');
                      return;
                    } 
                    
                    setGoodText(good);
```

Reply:
Still not fixed. Please feel free to ask for help in fe_chat